burn-common/src/benchmark.rs

```diff
@@ -167,10 +224,84 @@ where
         .output()
         .unwrap();
     let git_hash = String::from_utf8(output.stdout).unwrap().trim().to_string();
+    let durations = benchmark.run();
     BenchmarkResult {
-        timestamp,
+        raw: durations.clone(),
+        computed: BenchmarkComputations::new(&durations),
         git_hash,
         name: benchmark.name(),
-        durations: benchmark.run(),
+        options: benchmark.options(),
+        shapes: benchmark.shapes(),
+        timestamp,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use alloc::vec;
+
+    #[test]
+    fn test_min_max_median_durations_even_number_of_samples() {
+        let durations = BenchmarkDurations {
+            durations: vec![
+                Duration::new(10, 0),
+                Duration::new(20, 0),
+                Duration::new(30, 0),
+                Duration::new(40, 0),
+                Duration::new(50, 0),
+            ],
+        };
+        let (min, max, median) = durations.min_max_median_durations();
+        assert_eq!(min, Duration::from_secs(10));
+        assert_eq!(max, Duration::from_secs(50));
+        assert_eq!(median, Duration::from_secs(30));
+    }
+
+    #[test]
+    fn test_min_max_median_durations_odd_number_of_samples() {
+        let durations = BenchmarkDurations {
+            durations: vec![
+                Duration::new(18, 5),
+                Duration::new(20, 0),
+                Duration::new(30, 0),
+                Duration::new(40, 0),
+            ],
+        };
+        let (min, max, median) = durations.min_max_median_durations();
+        assert_eq!(min, Duration::from_nanos(18000000005_u64));
+        assert_eq!(max, Duration::from_secs(40));
+        assert_eq!(median, Duration::from_secs(30));
+    }
+
+    #[test]
+    fn test_mean_duration() {
+        let durations = BenchmarkDurations {
+            durations: vec![
+                Duration::new(10, 0),
+                Duration::new(20, 0),
+                Duration::new(30, 0),
+                Duration::new(40, 0),
+                Duration::new(50, 0),
+            ],
+        };
+        let mean = durations.mean_duration();
+        assert_eq!(mean, Duration::from_secs(30));
```

Comment:
That's a bit picky, but with these durations the mean is equal to the median. Maybe change them a bit so we can be sure we're finding the mean and not the median. You could remove the duration 50 and the mean should be 25.